### ISSUE TITLE
[Inductor] philox randn doesn't follow standard normal distribution

### ISSUE THREAD

**Issue by min-jean-cho** [pytorch/pytorch#91944]
### 🐛 Describe the bug

`Philox4_32` `randn` for cpu generates the same value (~1.75) for different `seed`. 
Hence, `Philox4_32` `randn` is (1) not random; (2) not follow standard normal distribution.

```cpp
#include <ATen/core/PhiloxRNGEngine.h>
#include <iostream>

int main() {
  for (int i = 1; i <= 1000; ++i) {
      at::Philox4_32 engine(i);
      std::cout << engine.randn(10) << std::endl; // 1.75668
  }
}
```

I think the root cause of (1) is due to resetting the state every time of `randn` https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/PhiloxRNGEngine.h#L108. Please help provide some context behind this call there.

The root cause of (2) is because `normalize_pair_uniform`  https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/PhiloxRNGEngine.h#L213-L226 is incorrect.

### Additional Context
This issue propagates to TorchInductor `randn` codegen defined with `Philox4_32` `randn` https://github.com/pytorch/pytorch/blob/master/torch/_inductor/codegen/cpp_prefix.h#L33-L36. See comment here https://github.com/pytorch/pytorch/pull/91207#discussion_r1065295810. 

```python
import torch

def foo(x):
    return torch.randn(x.size())

opt_foo = torch.compile(foo)
x = opt_foo(torch.randn(10000))
print(x)
print("mean: ", torch.mean(x)) # 1.7567, but E[X] should be close to 0
print("std: ", torch.std(x)) # 2.3843e-07 (i.e., is not random), but SD[X] should be close to 1

```

### Versions

torch==2.0.0a0+gitb8f35ec

cc @ezyang @gchanan @zou3519 @pbelevich @soumith @msaroufim @wconstab @ngimel @bdhirsh

**Comment by ngimel** [comment]
cc @voznesenskym
